generator/common/beatgen/setup/setup.go

```diff
@@ -18,111 +18,132 @@
-// Set the NEWBEAT_DEV env variable to use a slow `cp` copy that will catch uncommited changes
-func CopyVendor() error {
-	vendorPath := "./vendor/github.com/elastic/"
+func copyReplacedModules() error {
+	const goModPath = "go.mod"
+
 	beatPath, err := devtools.ElasticBeatsDir()
 	if err != nil {
-		return errors.Wrap(err, "Could not find ElasticBeatsDir")
+		return err
 	}
-	err = os.MkdirAll(vendorPath, 0755)
+
+	inMod, err := os.Open(filepath.Join(beatPath, goModPath))
 	if err != nil {
-		return errors.Wrap(err, "error creating vendor dir")
+		return err
 	}
+	defer inMod.Close()
 
-	isClean, err := checkBeatsDirClean()
-	if err != nil {
-		return errors.Wrap(err, "error in checkIfBeatsDirClean")
+	s := bufio.NewScanner(inMod)
+	inReplaceSection := false
+	replacedLines := []string{
+		"",
+		"replace (",
 	}
+	for s.Scan() {
+		if err := s.Err(); err != nil {
+			return err
+		}
 
-	if !isClean {
-		//Dev mode. Use CP.
-		fmt.Printf("You have uncommited changes in elastic/beats. Running CopyVendor running in dev mode, elastic/beats will be copied into the vendor directory with cp\n")
-		vendorPath = filepath.Join(vendorPath, "beats")
+		line := s.Text()
+		if inReplaceSection {
+			replacedLines = append(replacedLines, line)
 
-		err = sh.Run("cp", "-R", beatPath, vendorPath)
-		if err != nil {
-			return errors.Wrap(err, "error copying vendor dir")
+			if line == ")" {
+				break
+			}
+			continue
 		}
-		err = sh.Rm(filepath.Join(vendorPath, ".git"))
-		if err != nil {
-			return errors.Wrap(err, "error removing vendor git directory")
-		}
-		err = sh.Rm(filepath.Join(vendorPath, "x-pack"))
-		if err != nil {
-			return errors.Wrap(err, "error removing x-pack directory")
-		}
-	} else {
-		//not dev mode. Use git archive
-		vendorPath = filepath.Join(vendorPath, "beats")
-		err = os.MkdirAll(vendorPath, 0755)
-		if err != nil {
-			return errors.Wrap(err, "error creating vendor dir")
+
+		if line == "replace (" {
+			inReplaceSection = true
 		}
-		err = sh.Run("sh",
-			"-c",
-			"git archive --remote "+beatPath+" HEAD |  tar -x --exclude=x-pack -C "+vendorPath)
+	}
+
+	outMod, err := os.OpenFile(goModPath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return err
+	}
+	defer outMod.Close()
+
+	w := bufio.NewWriter(outMod)
+	for _, rep := range replacedLines {
+		_, err = w.WriteString(rep + "\n")
 		if err != nil {
-			return errors.Wrap(err, "error running git archive")
+			return err
 		}
 	}
 
-	return nil
+	return w.Flush()
+}
 
+func VendorModule() error {
+	return gotool.Mod.Vendor()
 }
 
-// checkIfBeatsDirClean checks to see if the working elastic/beats dir is modified.
-// If it is, we'll use a different method to copy beats to vendor/
-func checkBeatsDirClean() (bool, error) {
-	beatPath, err := devtools.ElasticBeatsDir()
+// CopyVendor copies a new version of beats into the vendor directory of PWD
+func CopyVendor() error {
```

Comment:
This target is also used in `mage vendorUpdate` to update vendor in existing beats, should this target be modified too to ask for what version of beats to update to?

Reply:
Selecting the desired Beats version should be done by `go get`. If the user wants to update beats, he/she has to run the following commands:
```sh
$ go get github.com/elastic/beats/v7@revision  # updates go.mod file
$ mage copyVendor                              # updates the vendor folder
```